uefi/src/table/mod.rs

```diff
@@ -37,42 +37,40 @@ pub unsafe fn set_system_table(ptr: *const uefi_raw::table::system::SystemTable)
 ///
 /// # Panics
 ///
-/// Panics if the system table has not been set with `set_system_table`, or if
-/// boot services are not available (e.g. if [`exit_boot_services`] has been
-/// called).
+/// Panics if the system table has not been set with [`set_system_table`].
 ///
 /// [`exit_boot_services`]: SystemTable::exit_boot_services
-pub fn system_table_boot() -> SystemTable<Boot> {
+pub fn system_table_boot() -> Option<SystemTable<Boot>> {
```

Comment:
This API is mostly an idea of @nicholasbishop , so I'd like to hear your opinon

Reply:
Seems reasonable. Since it now returns option, perhaps we should also have it return `None` if `SYSTEM_TABLE` is null instead of panicking.

Reply:
On the long term, we want the main function generated by the `#[entry]` macro to set the system table automatically. So I think, panicking is fine. This complexity should be hidden from users via an additional Option layer

Reply:
Yikes. Obsolete because of #1188